recipes/easylzma/all/conanfile.py

```diff
@@ -52,6 +51,10 @@ def source(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
+        # Relocatable shared libs on macOS
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        # Silence CMake warning about LOCATION property
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0026"] = "OLD"
```

Comment:
Please update the PR description and paste the error that you are getting - we don't know what motivates this change.

Reply:
There is no error, but as explained in PR title, it's a more efficient way to set `@rpath` (`CMAKE_POLICY_DEFAULT_CMP0042`) instead of `fix_apple_shared_install_name` which should be the last resort. For the warning, I've updated my recipe locally, I would have to get old revision, but it displays a warning about POLICY CMP0025 not being set.

```
CMake Warning (dev) at elzma/CMakeLists.txt:36 (GET_TARGET_PROPERTY):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target "elzma".  Use the
  target name directly with add_custom_command, or use the generator
  expression $<TARGET_FILE>, as appropriate.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at test/CMakeLists.txt:24 (GET_TARGET_PROPERTY):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target "easylzma_test".  Use
  the target name directly with add_custom_command, or use the generator
  expression $<TARGET_FILE>, as appropriate.
```